sop-extras/src/Data/SOP/Counting.hs

```diff
@@ -68,64 +54,31 @@ import           Data.SOP.Strict
 newtype Exactly xs a = Exactly { getExactly :: NP (K a) xs }
 
 -- | At most one value for each type level index
-data AtMost :: [Type] -> Type -> Type where
+type AtMost :: [Type] -> Type -> Type
```

Comment:
Do we get type inference problems when this is polykinded?

Reply:
Looks to me like it would be fine if `xs` were a list of polykinds